src/extractors/positiveFlux.ts

```diff
@@ -1,26 +1,25 @@
 import { normalizeToOne } from "../utilities";
 
 export default function ({
-  complexSpectrum,
-  previousComplexSpectrum,
+  ampSpectrum,
+  previousAmpSpectrum,
 }: {
-  complexSpectrum: { real: number[]; imag: number[] };
-  previousComplexSpectrum: { real: number[]; imag: number[] };
+  ampSpectrum: Float32Array;
+  previousAmpSpectrum: Float32Array;
 }): number {
-  if (!previousComplexSpectrum) {
+  if (!previousAmpSpectrum) {
     return 0;
   }
 
-  const normalizedRealComponent = normalizeToOne(complexSpectrum.real);
-  const previousNormalizedRealComponent = normalizeToOne(
-    previousComplexSpectrum.real
-  );
+  const normalizedMagnitudeSpectrum = normalizeToOne(ampSpectrum);
+  const previousNormalizedMagnitudeSpectrum =
+    normalizeToOne(previousAmpSpectrum);
 
   let sf = 0;
-  for (let i = 0; i < normalizedRealComponent.length; i++) {
+  for (let i = 0; i < normalizedMagnitudeSpectrum.length; i++) {
     let x =
-      Math.abs(previousNormalizedRealComponent[i]) -
-      Math.abs(normalizedRealComponent[i]);
+      Math.abs(normalizedMagnitudeSpectrum[i]) -
+      Math.abs(previousNormalizedMagnitudeSpectrum[i]);
     sf += Math.pow(Math.max(x, 0), 2);
```

Comment:
`Math.max` is now necessary, as the amplitude spectrum values are guaranteed to be positive.

Reply:
The amplitude spectrum values are all guaranteed to be positive, but the value at each position in first amplitude spectrum will be either greater than, equal to, or less than the corresponding value at the same position in the second amplitude spectrum. The idea with this max is to only care about increases in magnitude over time and ignore decreases. This is because if flux is being used for onset detection, decreases in magnitude aren't relevant.

Reply:
Right yeah, sorry - of course.